src/nix/profile-list.md

```diff
@@ -26,18 +28,22 @@ R""(
   # nix build github:edolstra/nix-warez/91f2ffee657bf834e4475865ae336e2379282d34?dir=blender#packages.x86_64-linux.default
   ```
 
-  will build the package with index 1 shown above.
+  will build the package `blender-bin` shown above.
 
 # Description
 
 This command shows what packages are currently installed in a
 profile. For each installed package, it shows the following
 information:
 
-* `Index`: An integer that can be used to unambiguously identify the
+* `Name`: A unique name used to unambiguously identify the
```

Comment:
How do we guarantee that names are unambiguous? A name like `packages.x86_64-linux.hello` isn't.

Reply:
Isnt that susceptible to the same issues this outlines: https://github.com/NixOS/nix/issues/7961?

Reply:
> Isnt that susceptible to the same issues this outlines: #7961?

Only for packages with the same name. #7961 is about _any_ package.

A decision about installing multiple packages with the same name can be made in a follow up PR. (Error? Replace? Interactive question? Allow duplicates?)

Reply:
Maybe it makes sense to use some portion of the store hash as an alternative suffix? e.g. hello-3bjqzx, hello-6r9h8

Reply:
That could also be an option.

Personally I'd like to avoid duplicate names eventually altogether by erroring upon install. I think these kinds of improvements/iterations should be a decision for a follow-up PR. The current implementation of this PR is simple and more or less in line with the current implementation on master (numbering the packages).

I see this PR as the most minimal step we can make towards using names for profile entries.